### ISSUE TITLE
500 Internal error (Redmine 3.2)

### ISSUE THREAD

**Issue by morgan134** [maxrossello/redmine_app_timesheets#73]
Hi,

After installing redmine_app_timesheets I got a Internal error message when I try to create a new project.

Environment:
  Redmine version             3.2.0.stable
  Ruby version                   2.0.0-p598 (2014-11-13) [x86_64-linux]
  Rails version                   4.2.5

And the logs:

Started GET "/projects/new" for 46.105.39.151 at 2015-12-23 16:05:44 +0100
Processing by ProjectsController#new as HTML
  Current user: admin (id=1)
  Rendered projects/new.html.erb within layouts/base (28.7ms)
Completed 500 Internal Server Error in 457ms (ActiveRecord: 31.7ms)

ActionView::Template::Error (private method `open' called for nil:NilClass):
    21: <% end %>
    22:
    23: <% if Redmine::VERSION.to_s >= '3.2' %>
    24: <% if @project.safe_attribute?('default_version_id') && (default_version_options = project_default_version_options(@project)).present? %>
    25:   <p><%= f.select :default_version_id, project_default_version_options(@project), :include_blank => true %></p>
    26: <% end %>
    27: <% end %>
  app/helpers/projects_helper.rb:91:in `project_default_version_options'
  app/views/projects/new.html.erb:4:in `block in _b4233c4b7a95ffddc7b15d2acb52a042'
  app/helpers/application_helper.rb:1030:in `labelled_form_for'
  app/views/projects/new.html.erb:3:in `_b4233c4b7a95ffddc7b15d2acb52a042'
  lib/redmine/sudo_mode.rb:63:in `sudo_mode'

After some research this issue appears just after having installed this plugin.

This issue can be easily reproduce by following these steps :

- Install Redmine 3.2.0
- Install redmine_app__space // At this step it works great
- Install redmine_app_timesheets // And after :(
- Go to: Administration page -> Projects -> new project

**Comment by maxrossello** [comment]
Hi,
still haven't had time to run Redmine 3.2. Sudo mode is something recent in Redmine.
Please stick to 3.0 for the moment.

**Comment by greyman888** [comment]
Hi Max,

I don't know if this helps but I think that the problem is related to this commit in Redmine 3.2:

https://github.com/redmine/redmine/commit/9178d4748f19adee08e5b9d58ea1176836249eb6

**Comment by skibinstein** [comment]
Hi,

Any update on this issue?

**Comment by pbaldassarre** [comment]
HI,
news on version 3.2 ?

**Comment by maxrossello** [comment]
I'm sorry guys, my company is sticking to older Redmine and I have no effort to spend right now. I would welcome any contribution.
Thanks

**Comment by sammcj** [comment]
totally understand @maxrossello, to put it out there - if anyone is able to help maintain, cleanup and keep this plugin up to date I might be able to donate a small amount of $$ towards the effort, I'd have to clear it pass my manager first - we use it every day but we are a non-profit, charitible organisation so I just have to check first. Let me know if anyone is interested.

**Comment by greyman888** [comment]
Hi. Sorry it took me so long to come back to this. I have submitted a PR to fix it.

The challenge was making sure that the PR fixes it without causing other problems. This time I used the Redmine test suite with some small hacks.

My test setup is a little out-of-date but OK I think:
Redmine version: 3.2.0
Ruby version: 2.2.1
Rails version: 4.2.5

- I created a new project in project.yml to act as the timesheet backing project (project id:7).
- I then added some code to timesheet_app_project_patch.rb where needed like this:

```
if Rails.env = "test"
    ts_project = 7
else
    ts_project = Setting.plugin_redmine_app_timesheets['project'].to_i
end
```
I then ran the Redmine test suite. All the tests that involve custom fields threw errors because TS_FIELD_NAME isn't defined but pretty much everything else passed. Considering this shouldn't even really work at all, I was pretty happy with the outcome.

I followed this up with manual testing in a running version of Redmine and I couldn't find any problems.

When I can make the time, I'll set up a Redmine 3.3 instance and test it again.
